tikv-client/src/main/java/com/pingcap/tikv/TiSession.java

```diff
@@ -45,12 +45,10 @@
 
   private static final Map<String, TiSession> sessionCachedMap = new HashMap<>();
 
-  // Since we create session as singleton now, configuration change will not
-  // reflect change
-  public static TiSession getInstance(TiConfiguration conf) {
+  public static TiSession getInstance(TiConfiguration conf, boolean recreate) {
```

Comment:
When recreate is true, it seems that sessionCacheMap is useless. Mind if we change the method's signature?

Reply:
It will update the session with the same key in sessionMap.

Reply:
ic